core/src/main/java/cz/xtf/core/waiting/helpers/ExponentialTimeBackoff.java

```diff
@@ -0,0 +1,115 @@
+package cz.xtf.core.waiting.helpers;
+
+
+import java.util.Random;
+
+/**
+ * Class provide support for exponential time backoff. This is convenient if you need to check some event periodically.
+ * The class provides {@code next} method that implements wait (blocking or non-blocking - returns true/false whether code
+ * can proceed).
+ *
+ * Class wait time grow exponentially up to {@code maxBackoffMillis} time: 1sec, 2 sec, 4 sec, 8 sec,...,{@code maxBackoffMillis}.
+ */
+public class ExponentialTimeBackoff {
+	private long lastNonBlockingTime;
+	private boolean blocking;
+	private long maxBackoffMillis;
+
+	private long wait;
+	private Random random;
+	private long nonBlockingWaitMillis;
+
+	private ExponentialTimeBackoff(Builder builder) {
+		blocking = builder.blocking;
+		maxBackoffMillis = builder.maxBackoffMillis;
+		wait = 1;
+		random = new Random(System.currentTimeMillis());
+		lastNonBlockingTime = System.currentTimeMillis();
+		nonBlockingWaitMillis = getNonBlockingWaitMillis();
+	}
+
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	/**
+	 * Exponential backoff wait.
+	 * If wait is blocking, {@link Thread#sleep} is called and true is always returned.
+	 * If wait is non-blocking, the method returns true if logic can proceed (enough time has elapsed)
+	 *
+	 * @return true if logic can proceed
+	 */
+	public boolean next() {
+		return blocking ? nextBlocking() : nextNonBlocking();
+	}
+
+	private boolean nextBlocking() {
+		try {
+			long wait = getNonBlockingWaitMillis();
```

Comment:
I suggest changing the variable name here, you're obscuring the class field `wait`.